### PR TITLE
Changed FilesystemSource.read to use open-uri.

### PR DESCRIPTION
**Pull request by Tornquist**
By switching from File.read to open-uri's open, support is added for remote files.  Now markdown files with image links and other remote links should generate properly.  I have only tested with a few sample files, but I immediately saw an improvement in the generation of valid epub files from source documents that threw error messages before. 

Existing functionality is still supported - all test cases passed locally.

### CODE REVIEW

**Comment by avdgaag** [comment]
Thanks for your PR. This is an interesting change. On the one hand, it is nice and simple and gives your some extra functionality. On the other hand, I'm not sure about the implications of suddenly supporting remote data sources. How should we deal with sluggish responses, or non-200 responses? Client-side caching? Authentication?

Conceptually, it makes a lot more sense to me to download all the required sources on my local disk and use Rpub to compile a book for me, rather than Rpub go out and fetch some stuff from the web for me. Personally, I would prefer a simple shell script doing some `wget`s.

Lastly, it's a `FilesystemSource` that reads remote stuff. That doesn't really make sense. If anything, we should probably create another kind of data source that knows how to read remote data.

So, in summary: I'd prefer a separate data source _like_ `FilesystemSource` that users can opt into, that knows how to deal with remote data. What do you think?

**Comment by Tornquist** [comment]
You're right.  Frankly, I didn't think at all about the implications of opening up remote sources that may not have valid references, or may not return 200.

For context, I'm currently converting websites to markdown and then pushing them through rpub.  This change allows remote images in the documents to be properly embedded in the files. 

Your comments about sluggish responses, authentication, etc.  They all lead would lead me to argue against this change.  Saying that, something is needed.  If you have valid markdown with remote references and push them through the current version of the rpub gem, invalid books are generated.  The process fails and the resulting documents are worthless.

I haven't dug through the gem too much, and just isolated the FilesystemSource as the place where the errors I saw were occurring.  I think something like a `FileSource` data source that directs objects through the normal `File.read` or the `open-uri` implementation based on a user flag could work. That way by default no users have to worry about remote content, but if they would like requests to be attempted, the option is open.

**Comment by avdgaag** [comment]
> If you have valid markdown with remote references and push them through the current version of the rpub gem, invalid books are generated. The process fails and the resulting documents are worthless.

A first step might be to detect this case and emit warnings, maybe?

> I think something like a FileSource data source that directs objects through the normal File.read or the open-uri implementation based on a user flag could work. That way by default no users have to worry about remote content, but if they would like requests to be attempted, the option is open.

Would you like to take a shot at building this?
